parsl/dataflow/dflow.py

```diff
@@ -152,7 +152,13 @@ def __init__(self, config=Config()):
             self.monitoring.send(MessageType.WORKFLOW_INFO,
                                  workflow_info)
 
-        checkpoints = self.load_checkpoints(config.checkpoint_files)
+        if config.checkpoint_files is not None:
+            checkpoints = self.load_checkpoints(config.checkpoint_files)
+        elif config.checkpoint_files is None and config.checkpoint_mode is not None:
+            checkpoints = self.load_checkpoints(get_all_checkpoints(self.run_dir))
+        else:
+            checkpoints = {}
```

Comment:
shouldn't this be `checkpoints = []` ?

Reply:
`load_checkpoints` returns a dict (of arguments->result mappings) - what this is defaulting to empty is that dict, not defaulting the list of what to load.

Reply:
This is correct, `load_checkpoints` returns the memoized task table.